cmd/ocm/gcp/create-wif-config.go

```diff
@@ -0,0 +1,515 @@
+			default:
+				fmt.Printf("Warning: %s is not a supported access type\n", serviceAccount.AccessMethod)
+			}
+		}
+		return nil
+	}
+
+	// Create service accounts
+	for _, serviceAccount := range wifOutput.Status.ServiceAccounts {
+		serviceAccountID := serviceAccount.GetId()
+		serviceAccountName := wifOutput.Spec.DisplayName + "-" + serviceAccountID
+		serviceAccountDesc := poolDescription + " for WIF config " + wifOutput.Spec.DisplayName
+
+		fmt.Println("Creating service account", serviceAccountID)
+		_, err := CreateServiceAccount(gcpClient, serviceAccountID, serviceAccountName, serviceAccountDesc, projectId, true)
+		if err != nil {
+			return errors.Wrap(err, "Failed to create IAM service account")
+		}
+		log.Printf("IAM service account %s created", serviceAccountID)
+	}
+
+	// Bind roles and grant access
+	for _, serviceAccount := range wifOutput.Status.ServiceAccounts {
+		serviceAccountID := serviceAccount.GetId()
+
+		fmt.Printf("\t\tBinding roles to %s\n", serviceAccount.Id)
+		for _, role := range serviceAccount.Roles {
+			if !role.Predefined {
+				fmt.Printf("Skipping role %q for service account %q as custom roles are not yet supported.", role.Id, serviceAccount.Id)
+				continue
+			}
+			err := gcpClient.BindRole(serviceAccountID, projectId, fmtRoleResourceId(role))
+			if err != nil {
+				panic(err)
+			}
+		}
+		fmt.Printf("\t\tRoles bound to %s\n", serviceAccount.Id)
+
+		fmt.Printf("\t\tGranting access to %s...\n", serviceAccount.Id)
+		switch serviceAccount.AccessMethod {
+		case "impersonate":
+			if err := gcpClient.AttachImpersonator(serviceAccount.Id, projectId, impersonatorServiceAccount); err != nil {
+				panic(err)
+			}
+		case "wif":
+			if err := gcpClient.AttachWorkloadIdentityPool(serviceAccount, wifOutput.Status.WorkloadIdentityPoolData.PoolId, projectId); err != nil {
+				panic(err)
+			}
+		default:
+			fmt.Printf("Warning: %s is not a supported access type\n", serviceAccount.AccessMethod)
+		}
+		fmt.Printf("\t\tAccess granted to %s\n", serviceAccount.Id)
+	}
+
+	return nil
+}
+
+func CreateServiceAccount(gcpClient gcp.GcpClient, svcAcctID, svcAcctName, svcAcctDescription, projectName string, allowExisting bool) (*adminpb.ServiceAccount, error) {
+	request := &adminpb.CreateServiceAccountRequest{
+		Name:      fmt.Sprintf("projects/%s", projectName),
+		AccountId: svcAcctID,
+		ServiceAccount: &adminpb.ServiceAccount{
+			DisplayName: svcAcctName,
+			Description: svcAcctDescription,
+		},
+	}
+	svcAcct, err := gcpClient.CreateServiceAccount(context.TODO(), request)
+	if err != nil {
+		pApiError, ok := err.(*apierror.APIError)
+		if ok {
+			if pApiError.GRPCStatus().Code() == codes.AlreadyExists && allowExisting {
+				return svcAcct, nil
+			}
+		}
+	}
+	return svcAcct, err
+}
+
+func generateServiceAccountID(serviceAccount models.ServiceAccount) string {
```

Comment:
Is this an unused helper? Maybe the backend should make use of this to populate the account ids within the wif_config object...

Reply:
I think I ended up changing the GetId() command from cmd/ocm/gcp/models/service_account.go instead